File: app/src/main/java/xyz/satoshicano/halalfoodchecker/activity/SplashActivity.java
package xyz.satoshicano.halalfoodchecker.activity;

import android.os.Bundle;

import xyz.satoshicano.halalfoodchecker.R;

import android.app.Activity;
import android.content.Intent;
import android.os.Handler;

public class SplashActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);setContentView(R.layout.activity_splash);
    }

    @Override
    public void onStart(){
        super.onStart();
        final Handler hdl = new Handler();
        hdl.postDelayed(new splashHandler(), 2000);
    }

    private class splashHandler implements Runnable {
        public void run() {
            Intent intent = new Intent(getApplication(), MainActivity.class);
            startActivity(intent);
            SplashActivity.this.finish();
        }
    }
}

File: app/src/main/java/xyz/satoshicano/halalfoodchecker/activity/SplashActivity.kt
package xyz.satoshicano.halalfoodchecker.activity

import android.os.Bundle

import xyz.satoshicano.halalfoodchecker.R

import android.app.Activity
import android.content.Intent
import android.os.Handler

class SplashActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)
    }

    public override fun onStart() {
        super.onStart()
        val hdl = Handler()
        hdl.postDelayed(splashHandler(), 2000)
    }

    private inner class splashHandler : Runnable {
        override fun run() {
            val intent = Intent(application, MainActivity::class.java)
            startActivity(intent)
            this@SplashActivity.finish()
        }
    }
}